support/cas-server-support-otp-mfa-core/src/main/java/org/apereo/cas/otp/web/flow/OneTimeTokenAccountSaveRegistrationAction.java

```diff
@@ -16,16 +21,41 @@
  * @author Misagh Moayyed
  * @since 5.0.0
  */
+@Slf4j
 @RequiredArgsConstructor
 public class OneTimeTokenAccountSaveRegistrationAction extends AbstractAction {
     private final OneTimeTokenCredentialRepository repository;
+    private final OneTimeTokenAccountValidator tokenValidator;
 
     @Override
     protected Event doExecute(final RequestContext requestContext) {
+        val uid = WebUtils.getAuthentication(requestContext).getPrincipal().getId();
+        val credential = WebUtils.getCredential(requestContext, OneTimeTokenCredential.class);
         val account = requestContext.getFlowScope().get("key", OneTimeTokenAccount.class);
 
-        val uid = WebUtils.getAuthentication(requestContext).getPrincipal().getId();
-        repository.save(uid, account.getSecretKey(), account.getValidationCode(), account.getScratchCodes());
-        return success();
+        int token;
+        try {
+            token = tokenValidator.parseToken(credential);
+        } catch (final IllegalArgumentException e) {
+            LOGGER.error("Unable to extract token from Credential [{}] for user [{}]",
+                    credential, uid);
+            return new EventFactorySupport().event(this, CasWebflowConstants.TRANSITION_ID_REGISTER);
+        }
```

Comment:
Dont catch unchecked exceptions. Either make the exception part of the contract, or return a value that indicates an error. Given your log message that ignores the exception, the latter is the better approach.